doc/task-hierarchy.md

```diff
@@ -0,0 +1,164 @@
+Task Type
+- Contextual Question Generation
+   - Generate questions based on given context e.g. a task to create a question based on a paragraph.
+- Answer Generation: Reading Comprehension
+- Answer Generation: Contextual Question Answering
```

Comment:
What is the difference between "Contextual Question Answering" and "Reading Comprehension"?

Reply:
Context= passage in "Reading Comprehension"
Context= single sentence/word in "Contextual Question Answering"